python/pyspark/sql/utils.py

```diff
@@ -77,6 +83,8 @@ def deco(*a, **kw):
                 raise QueryExecutionException(s.split(': ', 1)[1], stackTrace)
             if s.startswith('java.lang.IllegalArgumentException: '):
                 raise IllegalArgumentException(s.split(': ', 1)[1], stackTrace)
+            if s.startswith('java.lang.NoClassDefFoundError: '):
```

Comment:
`NoClassDefFoundError` rarely happens unless we miss something in `spark2RemovedClasses`. Hiding it makes us hard to figure out the real place quickly.